tests/testUtils.py

```diff
@@ -50,6 +51,65 @@ class BlockLogAction(EnumType):
 addEnum(BlockLogAction, "prune_transactions")
 
 ###########################################################################################
+
+class Timeout:
+
+    def __init__(self, value, sleep_interval=None, leeway=None):
+        self.toCalculate = True if value == -1 else False
+        if value is not None:
+            assert isinstance(value, (int))
+            assert value >= -1
+        if sleep_interval is not None:
+            assert isinstance(sleep_interval, (int))
+            assert sleep_interval > 0
+        self.value = value
+        self.sleep_interval = sleep_interval
+        self.leeway = leeway
+
+    def __str__(self):
+        append = "[calculated based on block production]" if self.toCalculate else ""
+        desc = None
+        if self.value is None:
+            desc = "defaulted"
+        elif self.value >= 0:
+            desc = "%d sec" % (self.value)
+        else:
+            desc = ""
+        return "Timeout %s%s" % (desc, append)
+
+    def convert(self, startBlockNum, endBlockNum):
+        if self.value is None or self.value != -1:
+            return
+
+        timeout = self.leeway if self.leeway is not None else 10
+        if (endBlockNum > startBlockNum):
+            # calculation is performing worst case (irreversible block progression) which at worst will waste 5 seconds
+            blocksPerWindow = 12
+            blockWindowsToWait = (endBlockNum - startBlockNum + blocksPerWindow - 1) / blocksPerWindow
+            secondsPerWindow = blocksPerWindow / 2
+            timeout += blockWindowsToWait * secondsPerWindow
+
+        self.value = timeout
+        if self.sleep_interval is None:
+            self.sleep_interval = max(timeout / 10, 1)
+
+    def sleepTime(self, default=None):
+        time = self.sleep_interval if self.sleep_interval is not None else default
```

Comment:
In the init, it is checking that if the sleep_interval is not None, that the value is appropriate for a positive amount of time.  Here it is letting you either:
indicate that your instance fall determines the sleep_interval (by passing to init)
calculate it based off of your calculated wait time
or else use the passed in default value

yes it is complicated, and I don't think we really even need a sleep_interval (I think it is always 3 seconds), but figured if I didn't put this in here, that you end up having people not use things because they need x and y, and z, and some methods give you x and y and some give you y and z.  I'm fine with whatever, but would rather do a separate cleanup/refactor task, but I don't think we will be given time to actually do it.

Reply:
Added a comment, since if we do leave it in, it at least needs that.

Reply:
If I read it right, the function could be rewritten as
```
def sleepTime(self, default=Timeout.default_interval_seconds):
    return self.sleep_interval  if self.sleep_interval else default
```

This makes the code a lot less convoluted.

Reply:
Or, if not, I say we just pull all of them out of everything and just use 3 seconds (or less if remaining time is less than that)

Reply:
Removed sleepTime from WaitSpec.